### ISSUE TITLE
error: The specified stream(s) 'source, 1080p60, best' could not be found.

### ISSUE THREAD

**Issue by stevek123** [bastimeyer/livestreamer-twitch-gui#323]
### Description
When I try to open the twitch.tv/twitchconlive stream, I get the following error:

> Found matching plugin twitch for URL twitch.tv/twitchconlive
> error: The specified stream(s) 'source, 1080p60, best' could not be found.
> Available streams: 360p30, 720p60_alt, 144p30, 480p30, 720p30, 720p60, 540p30, 240p30, audio

### Expected / Actual behavior
It should open the best version that is available - "720p60 - source" (which is not the same as 720p60)

### Reproduction steps

Try to open the Twitch Main Stage from the featured channels

### Environment details (operating system, etc.)

Windows 10 x64

### Comments, logs, screenshots, etc.

If you need one, let me know where can I find it

**Comment by bastimeyer** [comment]
This is an upstream error. See chrippa/livestreamer#1429

> 720p60_alt

It looks like Twitch is randomly renaming qualities again. It's the same issue as #283. 
See https://github.com/bastimeyer/livestreamer-twitch-gui/blob/v0.15.2/src/app/models/localstorage/Settings.js#L97-L120

I honestly don't want to fix upstream bugs here. I can only add a static list of known qualities to the GUI that livestreamer itself doesn't detect.

The issue has been fixed in [streamlink](https://github.com/streamlink/streamlink) (the livestreamer fork), and it's working fine.
```bash
$ streamlink -p mpv twitch.tv/twitchconlive best
[cli][info] Found matching plugin twitch for URL twitch.tv/twitchconlive
[cli][info] Available streams: 720p60_alt, audio, 144p30 (worst), 240p30, 360p30, 480p30, 540p30, 720p30, 720p60 (best)
[cli][info] Opening stream: 720p60 (hls)
```

In order to use streamlink with Livestreamer Twitch GUI, you need to build the application off the master branch, since `v0.15.2` doesn't validate streamlink yet. 8f7c0e3d79e92b2b431e9ecee58712b1ef6de441

Instructions can be found here:
https://github.com/bastimeyer/livestreamer-twitch-gui/blob/v0.15.2/CONTRIBUTING.md#developing-and-building

**Comment by stevek123** [comment]
Thank you for the detailed response!

**Comment by bastimeyer** [comment]
Let's keep this thread open. I don't want to see any duplicates from people who don't look for closed issues.

Maybe I could add all available stream qualities to the "source" quality option in descending order. This is of course not future proof, like the previous attempt(s), but it will ensure that all streams can be opened, even if "source" and "1080p60" (which is already a fallback) are missing.

**Comment by simoniz0r** [comment]
Looking forward to a new release to fix this issue.  For now, I have to set streams that fail to open to be on High quality instead of Source, and then they'll open at 720p30.  Building the application myself to work with streamlink seems like a bit more work than I want to do lol.

**Comment by bastimeyer** [comment]
Fixed in 426fe6780fd9097a6fe892121e4732798730e2da...
Custom qualities can now also be defined in the settings->livestreamer menu (advanced settings).

https://github.com/bastimeyer/livestreamer-twitch-gui/blob/426fe6780fd9097a6fe892121e4732798730e2da/src/app/models/LivestreamerQualities.js

**Comment by bastimeyer** [comment]
https://github.com/bastimeyer/livestreamer-twitch-gui/releases/latest

I'll leave this thread open for a bit, though...
